dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataverse/template/TemplateService.java

```diff
@@ -0,0 +1,141 @@
+package edu.harvard.iq.dataverse.dataverse.template;
+
+import edu.harvard.iq.dataverse.DataverseRequestServiceBean;
+import edu.harvard.iq.dataverse.EjbDataverseEngine;
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import edu.harvard.iq.dataverse.engine.command.impl.CreateTemplateCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.DeleteTemplateCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseTemplateRootCommand;
+import edu.harvard.iq.dataverse.persistence.dataset.Template;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import io.vavr.control.Try;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Stateless
+public class TemplateService {
+
+    private static final Logger logger = Logger.getLogger(TemplateService.class.getCanonicalName());
+
+    private EjbDataverseEngine engineService;
+    private DataverseRequestServiceBean dvRequestService;
+    private TemplateDao templateDao;
+
+    private LocalDateTime currentTime = LocalDateTime.now();
```

Comment:
This won't work as we expect. Current time will be the time when service is created. Either use `LocalDateTime.now()` in method directly and in tests use assertions like: template.getCreateTime() is after time that clone was executed and before clone was ended. Or use Clock - setter + field for clock instance; in method LocalDateTime.now(clock); in test setClock with fixed value